### PR TITLE
proof-producer: add output_artifacts library tests

### PR DESCRIPTION
**Pull request by oclaw**

### CODE REVIEW

File: proof-producer/tests/libs/output_artifacts/test_assignment_table_writer.cpp [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    OutputArtifacts artifacts;
+    artifacts.write_full = true;
+
+    auto mem_stream = write_table_to_stringstream(artifacts);
+
+    read_and_check_text_header(mem_stream);
+    for (auto row = 0; row < desc_.usable_rows_amount; row++) {
+
+        for (auto col = 0; col < table_.witnesses_amount(); col++) {
+            auto expected = row < table_.witness_column_size(row) ? table_.witness(col)[row] : 0;
```
Comment (akokoshn) [inline proof-producer/tests/libs/output_artifacts/test_assignment_table_writer.cpp:149]: may be use padded reference table? (no need the column size)

Comment (oclaw) [reply proof-producer/tests/libs/output_artifacts/test_assignment_table_writer.cpp]: yes, looks like redundant copypaste. Removed, thx

File: proof-producer/tests/libs/output_artifacts/test_assignment_table_writer.cpp [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+            auto str = read_stringified_field(mem_stream);
+            ASSERT_EQ(expected.data.str(str.size(), std::ios_base::hex), str) << "row: " << row << " col: " << col; 
+        }
+        read_rest_of_line(mem_stream);
+    }
+    ASSERT_EQ(mem_stream.tellg(), mem_stream.tellp());
+}
+
+
+TEST_F(AssignmentTableWriterTest, WritePartialColumns)
```
[the remark this answers is not in the document]
Comment (oclaw) [reply proof-producer/tests/libs/output_artifacts/test_assignment_table_writer.cpp:180]: Thought that printing only one table part may be kind of a corner case which is not fully checked in smoke test which writes multiple parts

[the remark this answers is not in the document]
Comment (oclaw) [reply proof-producer/tests/libs/output_artifacts/test_assignment_table_writer.cpp:180]: I generally prefer to have some test as "usage example"  which should be easy to read, and, separately, smoke test with complex inputs and corner case checks

**Review by AndreyMlashkin — APPROVED** [review]

**Review by akokoshn — APPROVED** [review]
